baseline.gradle

```diff
@@ -86,11 +86,17 @@ subprojects {
           '-Xep:CollectionUndefinedEquality:ERROR',
           // specific to Palantir - Uses name `log` but we use name `LOG`
           '-Xep:ConsistentLoggerName:OFF',
+          // TODO this is a recently added check. Figure out whether we adjust the code or suppress for good
```

Comment:
I believe we should probably address all of these and then configure them to `ERROR`.
Do you have a rough idea of how big the diff would be when addressing all of the new error prone checks? I think if it's not too much then we can include it in this PR.
Alternatively, we can open up separate issues and mark them as `good first issue` to give other people a chance to contribute here

Reply:
> Alternatively, we can open up separate issues and mark them as good first issue to give other people a chance to contribute here

I like that 👍

Reply:
Created https://github.com/apache/iceberg/issues/10853, https://github.com/apache/iceberg/issues/10854, https://github.com/apache/iceberg/issues/10855